src/components/results/shortcircuit/shortcircuit-analysis-result.tsx

```diff
@@ -9,67 +9,166 @@ import WaitingLoader from 'components/utils/waiting-loader';
-        ) as [ShortcircuitAnalysisResult, boolean, string];
+    const { onSortChanged, sortConfig } = useAgGridSort();
 
-    const isLoading =
-        analysisType === ShortcircuitAnalysisType.ALL_BUSES
-            ? isWaitingShortCircuitAnalysisResult
-            : oneBusShortcircuitAnalysisState === RunningStatus.RUNNING ||
-              isWaitingShortCircuitAnalysisResult;
+    // Effects
+    useEffect(() => {
+        setIsFetching(true);
+
+        const { colKey, sortWay } = sortConfig;
+
+        const sortKey = DATA_KEY_TO_SORT_KEY[colKey];
+        const sortWayValue = sortWay && (sortWay > 0 ? 'ASC' : 'DESC');
+
+        const parsedSortConfig =
+            sortKey && sortWayValue ? `${sortKey},${sortWayValue}` : '';
+
+        const selector = {
+            page,
+            size: rowsPerPage,
+            sort: parsedSortConfig,
+        };
+
+        const fetchAnalysisResult = (
+            fetchFunction: ShortCircuitAnalysisFetch
+        ) => {
+            fetchFunction(studyUuid, currentNode?.id)
+                .then((result: ShortcircuitAnalysisResult) => {
+                    const { content = [], totalElements } = result;
+                    setResult(content);
+                    setCount(totalElements);
+                })
+                .catch((error) =>
+                    snackError({
+                        messageTxt: error.message,
+                        headerId: intl.formatMessage({
+                            id: 'ShortCircuitAnalysisError',
+                        }),
+                    })
+                )
+                .finally(() => setIsFetching(false));
+        };
+
+        if (analysisType === ShortcircuitAnalysisType.ALL_BUSES) {
+            fetchAnalysisResult(
+                fetchShortCircuitAnalysisResult.bind(null, {
+                    studyUuid,
+                    currentNodeUuid: currentNode?.id,
+                    selector,
+                })
+            );
+        } else {
+            fetchAnalysisResult(
+                fetchOneBusShortCircuitAnalysisResult.bind(null, {
+                    studyUuid,
+                    currentNodeUuid: currentNode?.id,
+                })
+            );
+        }
+    }, [
+        page,
+        rowsPerPage,
+        studyUuid,
+        currentNode,
+        analysisType,
+        sortConfig,
+        intl,
+        snackError,
+    ]);
 
     return (
-        <WaitingLoader message={'LoadingRemoteData'} loading={isLoading}>
-            <ShortCircuitAnalysisResult result={shortCircuitAnalysisResult} />
-        </WaitingLoader>
+        <>
+            <WaitingLoader message={'LoadingRemoteData'} loading={isLoading}>
+                <ShortCircuitAnalysisResult
+                    result={result}
+                    onSortChanged={onSortChanged}
+                    sortConfig={sortConfig}
+                />
+            </WaitingLoader>
+            <CustomTablePagination
```

Comment:
There's a scrollbar on the right of this element. There's something to fix on CSS or else